examples/catvdog/transforms.py

```diff
@@ -125,70 +113,72 @@ def process(self, entity_id):
 
-        # return original entity ID for downstream processing
-        return entity_id
-
-    def input_data_exists(self, entity_id):
-        """Check if input data exists in GCS for a given entity ID.
+        # return original data for any downstream processing
+        yield data
 
-        Before invoking the `process` method with the received
-        `entity_id`, klio will call this method. If input data does not
-        exist, klio will trigger the configured parent job (if any).
 
-        Args:
-            entity_id (str): unique image file ID.
-        Returns
-            (bool): whether or not image exists in the configured
-                input bucket.
-        """
-        filename = "{}.jpg".format(entity_id)
-        filepath = os.path.join(self.input_loc, filename)
-        exists = self.gcs_client.exists(filepath)
-        return exists
+class CatVDogOutputCheck(beam.DoFn):
+    """Custom output data existence check to handle two output directories"""
 
-    def output_data_exists(self, entity_id):
-        """Check if output data exists in GCS for a given entity ID.
+    def setup(self):
+        """Setup instance variables that are not pickle-able"""
+        self.gcs_client = gcsio.GcsIO()
 
-        Before invoking the `process` method with the received
-        `entity_id`, klio will call this method. If output data already
-        exists and the `--force` flag from `klio message publish` was
-        not provided, klio will not invoke the `process` method and
-        effectively pass through the `entity_id` to the output pubsub
-        topic.
+    @decorators.handle_klio
+    def process(self, data):
+        """Detect if data for an element exists in one of two dirs in a bucket.
 
         Args:
-            entity_id (str): unique image file ID.
-        Returns
-            (bool): whether or not image exists in the configured
-                output bucket.
+            data (KlioMessage.data): data attribute of a KlioMessage including
+                fields ``element`` and ``payload``.
+        Returns:
+            apache_beam.pvalue.TaggedOutput: data tagged with either
+            ``found`` or ``not_found``.
         """
-        filename = "{}.jpg".format(entity_id)
-        dog_filepath = os.path.join(self.output_loc, "dog", filename)
-        if self.gcs_client.exists(dog_filepath):
-            return True
-
-        cat_filepath = os.path.join(self.output_loc, "cat", filename)
-        return self.gcs_client.exists(cat_filepath)
+        element = data.element.decode("utf-8")
+
+        oc = self._klio.config.job_config.data.outputs[0]
+        subdirs = ("cat", "dog")
+
+        outputs_exist = []
+
+        for subdir in subdirs:
+            path = f"{oc.location}/{subdir}/{element}{oc.file_suffix}"
+            self._klio.logger.info(f"Checking output in {path}")
+            exists = self.gcs_client.exists(path)
+            if exists:
```

Comment:
is this supposed to add something to `outputs_exist`?

Reply:
UGH good catch - accidentally deleted that line